.github/workflows/deploy.yml

```
@@ -0,0 +1,56 @@
+name: Deploy to shinyapps.io
+
+on:
+  push:
+    branches:
+      - CI-deployment # Adjust this as needed for your workflow
+
+jobs:
+  prepare-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Generate matrix from deployments.json
+        id: set-matrix
+        run: |
+          MATRIX_JSON=$(jq -c . < deployments.json)
+          echo "::set-output name=matrix::$MATRIX_JSON"
+
+  deploy:
+    needs: prepare-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.prepare-matrix.outputs.matrix)}}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.9.13"
+      - name: Cache deployment
+        id: cache-deployment
+        uses: actions/cache@v2
+        with:
+          path: ${{ matrix.folder }}
```

I believe there needs to be a step where we install the app packages.

Then we should also include the installed packages in the cache `path`.

```suggestion
          path: |
            ${{ matrix.folder }}
            INSTALL_PATH_HERE
```

IDK what the Install location is for the python packages

-----------------

Motivation: We cache both the shiny code and the app content

---

Oh that's a good point, what about including the shiny version in the requirements.txt file instead? I'll add an issue.